WalletWasabi.Gui/ViewModels/TextResourceViewModelBase.cs

```diff
@@ -30,28 +32,50 @@ public string Text
 			set => this.RaiseAndSetIfChanged(ref _text, value);
 		}
 
-		public Uri Target { get; }
+		public TextResource TextResource { get; }
 
 		private async Task<string> LoadDocumentAsync(Uri target)
 		{
 			var assetLocator = AvaloniaLocator.Current.GetService<IAssetLoader>();
-
 			using var stream = assetLocator.Open(target);
 			using var reader = new StreamReader(stream);
 			return await reader.ReadToEndAsync();
 		}
 
+		private async Task<string> LoadDocumentAsync(string path)
+		{
+			using var stream = File.OpenRead(path);
+			using var reader = new StreamReader(stream);
+			return await reader.ReadToEndAsync();
+		}
+
 		public override void OnOpen()
 		{
 			base.OnOpen();
 
 			Disposables = new CompositeDisposable();
-			LoadDocumentAsync(Target)
-				.ToObservable(RxApp.TaskpoolScheduler)
-				.Take(1)
-				.ObserveOn(RxApp.MainThreadScheduler)
-				.Subscribe(x => Text = x)
-				.DisposeWith(Disposables);
+			if (TextResource.HasContent)
+			{
+				Text = TextResource.Content;
+			}
+			else if (TextResource.HasAvaloniaTarget)
+			{
+				LoadDocumentAsync(TextResource.AvaloniaTarget)
```

Comment:
Is this mean if it is an embedded resource?

Reply:
That was the old way how we used it. I didn't want to remove it as `TextResourceViewModelBase` is a general control.

Reply:
ACK